### PR TITLE
eventing-channel feature | knative-plugin

### PR DESCRIPTION
**Pull request by SparshKesari**
**Description:**

- eventing-channel feature | knative-plugin

**Jira Id:** 

- [ODC-6682](https://issues.redhat.com/browse/ODC-6682) -  eventing-channel feature

**Pre-conditions/setup:**

**Checks for approving Epic scenarios Automation PR:**

- [x] Execute the @to-do tagged gherkin scripts manually
- [x] Convert the @to-do gherkin scripts to cypress automation scripts
- [x] Once scripts are automated, replace tag @to-do with @epic-number
- [x] Execute the scripts in Remote cluster

**Refactoring criteria for approving Automation PR:**

- [ ] update the test cases count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)

**Execution Commands:**
Example: 
```
export NO_HEADLESS=true && export CHROME_VERSION=$(/usr/bin/google-chrome-stable --version)
    BRIDGE_KUBEADMIN_PASSWORD=YH3jN-PRFT2-Q429c-5KQDr
    BRIDGE_BASE_ADDRESS=https://console-openshift-console.apps.dev-svc-4.8-042801.devcluster.openshift.com
    export BRIDGE_KUBEADMIN_PASSWORD
    export BRIDGE_BASE_ADDRESS
    oc login -u kubeadmin -p $BRIDGE_KUBEADMIN_PASSWORD
    oc apply -f ./frontend/integration-tests/data/htpasswd-secret.yaml
    oc patch oauths cluster --patch "$(cat ./frontend/integration-tests/data/patch-htpasswd.yaml)" --type=merge
    ./test-cypress.sh -p knative

```

**Screen shots**:

image

**Browser conformance**:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/knative-plugin/integration-tests/support/step-definitions/eventing/eventing-channel.ts [hunk 1]
```diff
@@ -16,11 +29,276 @@ When('user clicks on the Channel card', () => {
...
+When('user selects the {string} item', (applicationName: string) => {
+  cy.get(eventingPO.channel.dropDownMenu)
+    .contains(applicationName)
+    .click();
+});
+
+When('user adds the label', () => {
+  cy.get(eventingPO.channel.labelTagsInput)
+    .click()
+    .type('app.kubernetes.io/channel-label=12');
```
Comment (sanketpathak) [inline frontend/packages/knative-plugin/integration-tests/support/step-definitions/eventing/eventing-channel.ts:181]: Can this be provided from the feature file

File: frontend/packages/knative-plugin/integration-tests/support/step-definitions/eventing/eventing-channel.ts [hunk 2]
```diff
@@ -16,11 +29,276 @@ When('user clicks on the Channel card', () => {
...
+  cy.get(eventingPO.channel.keyAnnotations)
+    .last()
+    .type('eventing.knative.dev/channel.annotations');
+  cy.get(eventingPO.channel.valueAnnotations)
+    .last()
+    .type('test-annotation');
+});
+
+Then('user will see the newly added annotation', () => {
+  topologyPage.clickOnNode('ping-source');
```
Comment (sanketpathak) [inline frontend/packages/knative-plugin/integration-tests/support/step-definitions/eventing/eventing-channel.ts:214]: Is this step required?

File: frontend/packages/knative-plugin/integration-tests/support/step-definitions/eventing/eventing-channel.ts [hunk 3]
```diff
@@ -16,11 +29,276 @@ When('user clicks on the Channel card', () => {
...
+  app.waitForLoad();
+  perspective.switchTo(switchPerspective.Developer);
+  navigateTo(devNavigationMenu.Topology);
+  topologyHelper.verifyWorkloadDeleted(channelName);
+});
+
+Then('user will see the {string} tab', (tabName: string) => {
+  cy.get(eventingPO.channel.sidebarTab)
+    .contains(tabName)
+    .should('exist');
```
Comment (sanketpathak) [inline frontend/packages/knative-plugin/integration-tests/support/step-definitions/eventing/eventing-channel.ts:245]:
```suggestion
    .should('be.visible');
```

**Comment by hemantsaini-7** [comment]
/lgtm

image

**Review by sanketpathak — APPROVED** [review]

**Comment by sanketpathak** [comment]
Pr only affects our E2E tests

The pr only affects our E2E tests
/label qe-approved
/label docs-approved
/label px-approved

**Pull request merged**